packages/sanity/src/core/store/_legacy/document/document-pair/checkoutPair.test.ts

```diff
@@ -0,0 +1,317 @@
+import {beforeEach, expect, jest, test} from '@jest/globals'
+import {type SanityClient} from '@sanity/client'
+import {merge, of} from 'rxjs'
+import {delay} from 'rxjs/operators'
+
+import {checkoutPair} from './checkoutPair'
+
+const mockedDataRequest = jest.fn(() => of({}))
+const mockedObservableRequest = jest.fn((obj) => of(obj))
+
+const client = {
+  observable: {
+    listen: () => of({type: 'welcome'}).pipe(delay(0)),
+    getDocuments: (ids: string[]) =>
+      of([
+        {_id: ids[0], _type: 'any', _rev: 'any'},
+        {_id: ids[1], _type: 'any', _rev: 'any'},
+      ]),
+  },
+  dataRequest: mockedDataRequest,
+}
+
+const clientWithConfig = {
+  ...client,
+  withConfig: () => ({
+    ...client,
+    observable: {...client.observable, request: mockedObservableRequest},
+  }),
+  config: () => ({dataset: 'production'}),
```

Comment:
should the dataset be hardcoded or should we get this from some environment variable or CI config?

Reply:
It would be nice to standardize it... I'm not sure we've set up an affordance for that -- mockClient datasets seem to be all over the shop and hardcoded in our tests:

- https://github.com/sanity-io/sanity/blob/b7961fdb5985b48c0446c8636f0a0ca2124e0641/packages/sanity/src/core/store/_legacy/grants/__tests__/createGrantsStore.test.ts#L13
- https://github.com/sanity-io/sanity/blob/b7961fdb5985b48c0446c8636f0a0ca2124e0641/packages/%40sanity/util/src/client/concurrency-limiter/__test__/createClientConcurrencyLimiter.test.ts#L74
- https://github.com/sanity-io/sanity/blob/b7961fdb5985b48c0446c8636f0a0ca2124e0641/packages/sanity/src/core/studio/components/navbar/search/datastores/recentSearches.test.tsx#L55

I wonder if this falls under broader "remove tech debt" improvements.